nomad/lock/delay.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package lock
+
+import (
+	"sync"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/nomad/helper"
+)
+
+// DelayTimer is used to mark certain locks as unacquirable. When a locks TTL
+// expires, it is subject to the LockDelay configured within the variable. This
+// prevents another caller from acquiring the lock for some period of time as a
+// protection against split-brains. This is inspired by the lock-delay in
+// Chubby.
+type DelayTimer struct {
+
+	// delayTimers has the set of active delay expiration times, organized by
+	// ID, which the caller dictates when adding entries. The lock should be
+	// used for all access.
+	delayTimers map[string]time.Time
+	lock        sync.RWMutex
+}
+
+// NewDelayTimer returns a new delay timer manager.
+func NewDelayTimer() *DelayTimer {
+	return &DelayTimer{
+		delayTimers: make(map[string]time.Time),
```

Comment:
Wouldn't it be good to initialize the mutex in the constructor, too? I would expect trying to get a lock on an uninitialized mutex to cause a panic.